docs: clarify coprocessor auth stages (#4306)
apollographql/router: docs/source/customizations/coprocessor.mdx

## Adding authorization claims via coprocessor

> To use the [authorization directives](../configuration/authorization#authorization-directives), a request needs to include **claims**—the details of its authentication and scope. The most straightforward way to add claims is with [JWT authentication](../configuration/./authn-jwt). You can also add claims with a [`RouterService` or `SupergraphService` coprocessor](#how-it-works) since it hooks into the request lifecycle before the router applies authorization logic.

Reviewer: ```suggestion
To use the [authorization directives](../configuration/authorization#authorization-directives), a request needs to include **claims**—the details of its authentication and scope. The most straightforward way to add claims is with [JWT authentication](../configuration/./authn-jwt). You can also add claims with a [`RouterService` or `SupergraphService` coprocessor](#how-it-works) since they hook into the request lifecycle before the router applies authorization logic.
```

> The router configuration necessary to call the coprocessor for authorization claims looks like this:

Reviewer: ```suggestion
The router configuration needs to call the coprocessor for authorization claims that look like this:
```

Author: Hmm your suggestion doesn't quite make sense to me. I agree that my sentence isn't great either tho

Reviewer 2: How about:
```
When processing authorization claims, the required router configuration looks like this:
```

Reviewer: ```suggestion
An example configuration of the router calling a coprocessor for authorization claims:
```

```yaml title="router.yaml"
coprocessor:
  url: http://127.0.0.1:8081 # Required. Replace with the URL of your coprocessor's HTTP endpoint.

>   router: # By including this key, a coprocessor can hook into the `RouterService`. You can also use `supergraph` for authorization.

Reviewer: ```suggestion
  router: # By including this key, a coprocessor can hook into the `RouterService`. You can also use `SupergraphService` for authorization.
```
